kuba-kostrzewski/simpleToDoApp/src/main/java/pl/kostrzej/simpleToDoApp/components/task/TaskConsoleController.java

```diff
@@ -0,0 +1,148 @@
+            title = scanner.nextLine();
+        } while (fieldValidator.isFieldEmpty(title, "tytuł"));
+        task.setTitle(title);
+        System.out.println("Opis: ");
+        System.out.println(task.getDescription());
+        System.out.println("Wprowadź nowy opis lub zostaw puste: ");
+        task.setDescription(scanner.nextLine());
+        System.out.println("Data: ");
+        System.out.println(task.getDate());
+        System.out.println("Wprowadź nową datę: ");
+        task.setDate(dateController.readDate());
+        taskService.saveTask(task);
+        log.info("Task edited successfully.");
+        return task;
+    }
+    @Override
+    public Task getTaskFromList(List<Task> tasks){
+        Integer taskIndex = null;
+        showAllTasks(tasks);
+        if (tasks.size()>0){
+            System.out.println("Podaj nr zadania:");
+            try {
+                taskIndex = readTaskIndex(tasks);
+                return tasks.get(taskIndex);
+            } catch (InputMismatchException e){
+                log.info("Invalid data type");
+                System.out.println("Wpisano niewłaściwe dane! Podaj liczbę!");
+            } catch (IndexOutOfBoundsException e){
+                System.err.println(e.getMessage());
+                log.info("Invalid task number! Task list size= {} Value typed to console = {} {}",
+                        tasks.size(), taskIndex, e.getClass().toString());
+            }
+        }
+        return null;
+    }
+    @Override
+    public void deleteTask(Task task) {
+        log.info("Deleting task process initialized.");
+        taskService.deleteTask(task);
+        log.info("Task deleted successfully.");
+    }
+
+    @Override
+    public Task changeTaskStatus(Task task) {
+        log.info("Changing task status process initialized.");
+            System.out.println("Czy chcesz zmienić status? (Y/N)");
+            try {
+                boolean answer = answerYes();
+                if (answer && task.getStatus().equals(TaskStatus.DONE)){
+                    task.setStatus(TaskStatus.UNDONE);
+                    log.info("Task status changed. Status: {}", task.getStatus());
+                    taskService.saveTask(task);
+                } else if (answer && task.getStatus().equals(TaskStatus.UNDONE)){
+                    task.setStatus(TaskStatus.DONE);
+                    log.info("Task status changed. Status: {}", task.getStatus());
+                    taskService.saveTask(task);
+                }
+            } catch (InvalidAnswerException e){
+                System.err.println(e.getMessage());
+                log.info("Invalid answer.");
+            }
+            log.info("Task status not changed.");
+    return task;
+    }
+
+    @Override
+    public void showAllTasks(List<Task> tasks){
+        log.info("Show all tasks process initialized.");
+        System.out.println("Lista zadań:");
+        if (tasks.isEmpty() || tasks == null){
```

Comment:
Nawet jeśli tasks będzie nullem to i tak otrzymasz NPE ze względu, że najpierw wykorzystujesz isEmpty.

Reply:
poprawione